internal/konnect/client.go

```diff
@@ -0,0 +1,169 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	if httpResp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("non-success response code from Koko: %d, resp body: %s", httpResp.StatusCode, string(respBuf))
+		// TODO: parse returned body to return a more detailed error
+	}
+
+	resp := &CreateNodeResponse{}
+	err = json.Unmarshal(respBuf, resp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON body: %v", err)
+	}
+
+	return resp, nil
+}
+
+func (c *KonnectAdminClient) UpdateNode(nodeID string, req *UpdateNodeRequest) (*UpdateNodeResponse, error) {
+	buf, err := json.Marshal(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal update node request: %v", err)
+	}
+	reqReader := bytes.NewReader(buf)
+	url := fmt.Sprintf("https://%s/kic/api/runtime_groups/%s/kic-nodes/%s", c.Address, c.RuntimeGroupID, nodeID)
+	httpReq, err := http.NewRequest("PUT", url, reqReader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request:%v", err)
+	}
+	httpResp, err := c.Client.Do(httpReq)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response: %v", err)
+	}
+	defer httpResp.Body.Close()
+
+	respBuf, err := io.ReadAll(httpResp.Body)
+	if err != nil {
+		err := fmt.Errorf("failed to read response body: %v", err)
+		return nil, err
+	}
+
+	if httpResp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("non-success response code from Koko: %d, resp body %s", httpResp.StatusCode, string(respBuf))
+	}
+
+	resp := &UpdateNodeResponse{}
+	err = json.Unmarshal(respBuf, resp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSON body: %v", err)
+	}
+	return resp, nil
+}
+
+func (c *KonnectAdminClient) ListNodes() (*ListNodeResponse, error) {
+	url := fmt.Sprintf("https://%s/kic/api/runtime_groups/%s/kic-nodes", c.Address, c.RuntimeGroupID)
+	httpResp, err := c.Client.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response: %v", err)
+	}
+
+	defer httpResp.Body.Close()
+
+	respBuf, err := io.ReadAll(httpResp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %v", err)
+	}
+
+	if httpResp.StatusCode/100 != 2 {
+		return nil, fmt.Errorf("non-success response from Koko: %d, resp body %s", httpResp.StatusCode, string(respBuf))
+	}
+
+	resp := &ListNodeResponse{}
+	err = json.Unmarshal(respBuf, resp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %v", err)
+	}
+	return resp, nil
+}
+
+func (c *KonnectAdminClient) DeleteNode(nodeID string) error {
+	url := fmt.Sprintf("https://%s/kic/api/runtime_groups/%s/kic-nodes/%s", c.Address, c.RuntimeGroupID, nodeID)
+	httpReq, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return fmt.Errorf("failed to create request:%v", err)
+	}
+	httpResp, err := c.Client.Do(httpReq)
+	if err != nil {
+		return fmt.Errorf("failed to get response: %v", err)
+
+	}
+	defer httpResp.Body.Close()
+
+	if httpResp.StatusCode/100 != 2 {
```

Comment:
😱 How about we just do (not sure about the 3xx codes though, you can change it to match on 4xx instead)

```suggestion
	if httpResp.StatusCode < 200 || httpResp.StatusCode >= 300 { 
```